tiledb/sm/misc/mgc_dict.cc

```diff
@@ -37,45 +39,30 @@
 namespace tiledb {
 namespace sm {
 
-static const char magic_mgc_compressed_bytes[] = {
-#include "magic_mgc_gzipped.bin"
-};
+const tiledb::sm::ByteVecValue prepare_data() {
+  tiledb::sm::ByteVecValue expanded_buffer(magic_mgc_decompressed_size);
 
-shared_ptr<tiledb::sm::ByteVecValue> magic_dict::expanded_buffer_;
+  ConstBuffer input(magic_mgc_compressed_bytes, magic_mgc_compressed_size);
+  PreallocatedBuffer output(expanded_buffer.data(), expanded_buffer.size());
+  ZStd::ZSTD_Decompress_Context ctx;
+  ZStd::decompress(ctx, &input, &output);
 
-void* magic_dict::uncompressed_magic_dict_ = nullptr;
-
-static magic_dict magic_dict_object;
-
-magic_dict::magic_dict() {
-}
-
-void magic_dict::prepare_data() {
-  if (uncompressed_magic_dict_)
-    return;
-
-  expanded_buffer_ = make_shared<tiledb::sm::ByteVecValue>(HERE());
-  gzip_decompress(
-      expanded_buffer_,
-      reinterpret_cast<const uint8_t*>(&magic_mgc_compressed_bytes[0]));
-
-  uncompressed_magic_dict_ = expanded_buffer_.get()->data();
+  return expanded_buffer;
 }
 
 int magic_dict::magic_mgc_embedded_load(magic_t magic) {
-  if (!uncompressed_magic_dict_)
-    prepare_data();
+  auto& buffer = expanded_buffer();
 
-  void* data[1] = {uncompressed_magic_dict_};
-  size_t sizes[1] = {expanded_buffer_->size()};
+  void* data = const_cast<unsigned char*>(buffer.data());
+  size_t size = buffer.size();
   // zero ok, non-zero error
-  return magic_load_buffers(magic, &data[0], &sizes[0], 1);
+  return magic_load_buffers(magic, &data, &size, 1);
 }
 
-const shared_ptr<tiledb::sm::ByteVecValue> magic_dict::expanded_buffer() {
-  if (!uncompressed_magic_dict_)
-    prepare_data();
-  return expanded_buffer_;
+const tiledb::sm::ByteVecValue& magic_dict::expanded_buffer() {
```

Comment:
The return value involving `ByteVecValue` was never a good idea for this function. It's only use is for testing, in addition. It should probably just return `span`.

In addition it would be better to declare `prepare_data` in the header and to move this function to the test code.

Reply:
Changed to return a span. I don't understand why we should export `prepare_data`.